app/src/main/java/com/moyerun/moyeorun_android/views/BadgeRoundImageView.kt

```diff
@@ -0,0 +1,89 @@
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.moyerun.moyeorun_android.R
+import com.moyerun.moyeorun_android.databinding.ViewBadgeRoundImageBinding
+
+class BadgeRoundImageView : ConstraintLayout {
+    constructor(context: Context): super(context) { initView() }
+    constructor(context: Context, attrs: AttributeSet): super(context, attrs) {
+        initView()
+        setAttrs(attrs)
+    }
+    constructor(context: Context, attrs: AttributeSet, defStyle: Int): super(context, attrs) {
+        initView()
+        setAttrs(attrs, defStyle)
+    }
+
+    private lateinit var binding: ViewBadgeRoundImageBinding
+
+    private fun initView() { binding = ViewBadgeRoundImageBinding.inflate(LayoutInflater.from(context), this, true) }
+
+    private fun getAttrs(attributeSet: AttributeSet, customViewAttrs: IntArray,  defStyle: Int=0, defStyleRes: Int=0)
+    = context.obtainStyledAttributes(attributeSet, customViewAttrs, defStyle, defStyleRes)
+
+    private fun setAttrs(attrs: AttributeSet) {
+        val attributesTypedArray = getAttrs(attrs, R.styleable.BadgeRoundImageView)
+        setTypedArray(attributesTypedArray)
+    }
+    private fun setAttrs(attrs: AttributeSet, defStyle: Int) {
+        val attributesTypedArray = getAttrs(attrs, R.styleable.BadgeRoundImageView, defStyle, 0)
+        setTypedArray(attributesTypedArray)
+    }
+    private fun setTypedArray(typedArray: TypedArray) {
+        binding.imgBigCircle.apply {
+            val bigCircleImgResId = typedArray.getResourceId(R.styleable.BadgeRoundImageView_bigCircleImgSrc, R.drawable.ic_launcher_foreground)
+            val bigCircleImgBgResId = typedArray.getResourceId(R.styleable.BadgeRoundImageView_bigCircleBackgroundColor, R.drawable.ic_launcher_foreground)
+            val bigCircleSize = typedArray.getDimensionPixelSize(R.styleable.BadgeRoundImageView_bigCircleSize, 0)
+
+            this.setImageResource(bigCircleImgResId)
+            this.setBackgroundResource(bigCircleImgBgResId)
+            this.layoutParams.apply {
+                this.height = bigCircleSize
+                this.width = bigCircleSize
+            }
+        }
+
+        binding.imgBadgeSymbol.apply {
+            val badgeResId = typedArray.getResourceId(R.styleable.BadgeRoundImageView_badgeImgSrc, R.drawable.ic_launcher_foreground)
+            val badgeBgResId = typedArray.getResourceId(R.styleable.BadgeRoundImageView_badgeBackgroundColor, R.drawable.ic_launcher_foreground)
+            val badgeSize = typedArray.getDimensionPixelSize(R.styleable.BadgeRoundImageView_badgeSize, 0)
+
+            this.setImageResource(badgeResId)
+            this.setBackgroundResource(badgeBgResId)
+            this.layoutParams.apply {
+                this.height = badgeSize
+                this.width = badgeSize
+            }
+        }
+        typedArray.recycle()
+    }
+
+    fun setBigCircleImgSrc(imgResId: Int) {
+        binding.imgBigCircle.setImageResource(imgResId)
```

Comment:
파라미터 타입이 `Int`인데, 이러면 아무 정수형이나 다 들어올 수 있어요. 나중에 앱 터지고서야 잘못 넣었음을 알게됩니다.
`@DrawableRes` 어노테이션을 달아주는 게 좋아보입니다. 이런식으로 파라미터를 제한하는 어노테이션이 많으니 참고하셔요

```suggestion
    fun setBigCircleImgSrc(@DrawableRes imgResId: Int) {
        binding.imgBigCircle.setImageResource(imgResId)
    }
```

Reply:
넵ㅎㅎ 어노테이션을 달아주는게 좋을 것 같네요! 수정해놓도록 하겠습니다~